cmd/schedule-builder: Add CLI tool for release schedule generation (#1313)
kubernetes/release: cmd/schedule-builder/README.md

# Schedule Builder

> This simple tool have the objective parsing the schedule yaml file located in [SIG-Release](https://github.com/kubernetes/sig-release/blob/master/releases/) where shows in machine readable format the scheduled releases and past patch releases of the current Kubernetes Releases.

Reviewer: ```suggestion
This simple tool has the objective to parse the yaml file located in [SIG-Release](https://github.com/kubernetes/sig-release/blob/master/releases/), which shows the scheduled and past patch releases of the current Kubernetes Release cycle in machine readable format.
```

Author: done

## Install

The simplest way to install the `schedule-builder` CLI is via `go get`:

```

> GO111MODULE=on go get k8s.io/release/cmd/schedule-builder

Reviewer: I think it works also without setting the env var:

```suggestion
$ go get k8s.io/release/cmd/schedule-builder
```

Author: done

| 1.16.8        | 2020-03-09                                                                            | 2020-03-12  |
| 1.16.7        | 2020-02-07                                                                            | 2020-02-11  |
| 1.16.6        | No-op release https://groups.google.com/d/topic/kubernetes-dev/Mhpx-loSBns/discussion | 2020-01-21  |
| 1.16.5        | 2020-01-10                                                                            | 2020-01-14  |
| 1.16.4        | 2019-12-06                                                                            | 2019-12-11  |
| 1.16.3        | 2019-11-08                                                                            | 2019-11-13  |
| 1.16.2        | 2019-10-11                                                                            | 2019-10-15  |
| 1.16.1        | 2019-09-27                                                                            | 2019-10-02  |
```

Also can save the schedule in a file, to do that, you can set the `--output-file` flag together with the filename.

```
$ schedule-builder --schedule-path ../sig-release/releases/schedule.yaml --output-file my-schedule.md
```

> ## Building From Source

Reviewer: I would move that section below `## Install` and before `## Usage`

Author: done

To build the `schedule-builder` tool, check out this repo to your `$GOPATH`:

```

> git clone git@github.com:kubernetes/release.git $(go env GOPATH)/src/k8s.io/release

Reviewer: Let’s either prefix all shell commands with `$ ` or none of them. :)

Author: done

```

> Run the following from the root of the repository to build the `gschedule-builderh2gcs` binary:

Reviewer: Do you mean “…the `schedule-builder` binary“?

Author: err.. yes :)

Author: done

```

> bazel build //cmd/schedule-builder

Reviewer: Do we really want to recommend bazel builds? I suggest we stick to `go build/run/install` or the `compile-release-tools` script.

Author: just following the same as in the other readme. but i agree, will remove
